src/cljs/ataru/hakija/subs.cljs

```diff
@@ -35,6 +35,27 @@
       (get-in db [:form :selected-language])
       :fi))) ; When user lands on the page, there isn't any language set until the form is loaded)
 
+(defn- selected-hakukohteet [db]
+  (map :value (get-in db [:application :answers :hakukohteet :values] [])))
+
+(defn- selected-hakukohteet-from-tarjonta [db]
+  (let [selected-hakukohteet     (set (selected-hakukohteet db))]
+    (->> (get-in db [:form :tarjonta :hakukohteet])
+         (filter #(contains? selected-hakukohteet (:oid %))))))
+
+(re-frame/reg-sub
+  :application/selected-hakukohteet-for-field
+  (fn [db [_ field]]
+    (when-let [ids (seq (concat (get field :belongs-to-hakukohderyhma [])
+                                (get field :belongs-to-hakukohteet [])))]
+      (let [selected-hakukohteet-from-tarjonta (selected-hakukohteet-from-tarjonta db)]
+        (->> selected-hakukohteet-from-tarjonta
+             (filter #(not (empty? (clojure.set/intersection
+                                    (set ids)
+                                    (set (concat [(:oid %)] (get % :hakukohderyhmat [])))))))
+             (map :oid)
```

Comment:
`selected-hakukohteet-for-field` vois mieluummin palauttaa koko hakukohde oliot. `question-hakukohde-names` nimittäin joutuu heti kaivamaan ne uudestaan oidien perusteella